python/_spglib.c

```diff
@@ -869,6 +874,54 @@ static PyObject *py_get_magnetic_spacegroup_type(PyObject *self,
     return array;
 }
 
+static PyObject *py_get_magnetic_spacegroup_type_from_symmetry(PyObject *self,
+                                                               PyObject *args) {
+    PyArrayObject *py_rotations;
+    PyArrayObject *py_translations;
+    PyArrayObject *py_time_reversals;
+    PyArrayObject *py_lattice;
+    double symprec;
+
+    int(*rotations)[3][3];
+    double(*translations)[3];
+    int *time_reversals;
+    double(*lattice)[3];
+    int num_operations;
+
+    SpglibMagneticSpacegroupType msg_type;
+
+    int n;
+    PyObject *array;
+
+    if (!PyArg_ParseTuple(args, "OOOOd", &py_rotations, &py_translations,
+                          &py_time_reversals, &py_lattice, &symprec)) {
+        return NULL;
+    }
+    rotations = (int(*)[3][3])PyArray_DATA(py_rotations);
+    translations = (double(*)[3])PyArray_DATA(py_translations);
+    time_reversals = (int *)PyArray_DATA(py_time_reversals);
+    lattice = (double(*)[3])PyArray_DATA(py_lattice);
+    num_operations = PyArray_DIMS(py_rotations)[0];
+
+    msg_type = spg_get_magnetic_spacegroup_type_from_symmetry(
```

Comment:
There are a few interfaces that do not properly set error when they fail. Can you check them?